pkg/provisioningv2/rke2/planner/config.go

```diff
@@ -114,7 +114,12 @@ func addRoleConfig(config map[string]interface{}, controlPlane *rkev1.RKEControl
 	}
 
 	if sdr := settings.SystemDefaultRegistry.Get(); sdr != "" && !isOnlyWorker(entry) {
-		config["system-default-registry"] = sdr
+		// only pass the global system-default-registry if we have not defined a different registry for system-images within the UI.
```

Comment:
Please correct me if I am wrong: 
the cluster-level registry is used for all images, including the user's workloads. 
```suggestion
		// only pass the global system-default-registry if we have not defined a different registry under the cluster configuration within the UI.
```

Reply:
User workloads can come from the cluster level registry, but are not limited to the cluster level registry. The `registries.yaml` ensures that, when the FQDN of the cluster level registry is provided, no `imagePullSecrets` will need to be defined for the workload to be pulled successfully.